crates/runtime/src/lib.rs

```diff
@@ -304,14 +333,18 @@ impl Runtime {
     }
 
     pub async fn start_pods_watcher(&mut self) -> notify::Result<()> {
-        let mut current_app = Arc::clone(&self.app);
-
         let mut rx = self.pods_watcher.watch()?;
 
         while let Some(new_app) = rx.recv().await {
+            let current_app = self.app.read().await;
+            let current_datasets = current_app.datasets.clone();
```

Comment:
Once you acquire the write lock you don't have to clone the data anymore.

Reply:
Updated, thanks!